File: matlab/SIRscript.m
%
% File:   SIRscript.m  (for use with Matlab)
% Author: Warren Weckesser
%         Department of Mathematics
%         Colgate University
%
% Copyright (c) 2005 Warren Weckesser
%
%
% Set the parameters
%
r = 0.12;
g = 0.07;
%
% Set the initial conditions
%
S0 = 0.995;
I0 = 0.005;
R0 = 0.0;
%
% Put the initial conditions into the vector x0
%
x0 = [S0; I0; R0];
%
% tinfal is the stop time of ODE solver.
%
tfinal = 250.0;
%
% Call the ode solver
%
[t,x] = ode45(@SIRvectorfield,[0 tfinal],x0,[],r,g);
%
% Plot the solution (as functions of t)
%
clf;
plot(t,x(:,1),t,x(:,2),t,x(:,3));
tstr = sprintf('Solution of the SIR Model  (r=%.2f, gamma=%.2f)',r,g);
title(tstr);
xlabel('t');
legend('S','I','R');

File: python/SIRscript.py
#
# File:  SIRscript.py
# Author: Warren Weckesser
#
# Copyright(c) 2005, 2006, 2019 Warren Weckesser
#

import numpy as np
from scipy.integrate import odeint
import matplotlib.pyplot as plt

#
# Get the SIR vector field function from SIRvectorfield.py.
#
from SIRvectorfield import SIRvectorfield

#
# Set the parameters.
#
r = 0.12
g = 0.07

#
# Set the initial conditions and put into the array x0.
#
S0 = 0.995
I0 = 0.005
R0 = 0.0
x0 = [S0, I0, R0]

#
# Create the time samples for the output of the
# ODE solver.
#
tfinal = 250.0
t = np.arange(0.0, tfinal, tfinal/200)

#
# Call the ODE solver.
#
x = odeint(SIRvectorfield, x0, t, args=(r, g))

outdata = np.zeros((len(t),4))
outdata[:,0] = t
outdata[:,1:] = x
np.savetxt('SIRdata.txt', outdata)

plt.plot(t, x[:, 0], '-', label='S')
plt.plot(t, x[:, 1], '--', label='I')
plt.plot(t, x[:, 2], ':', label='R')

plt.xlabel('t')
plt.title('Solution of the SIR Model (r=%s, gamma=%s)' % (r, g))
plt.legend(shadow=True)
plt.grid()
plt.savefig('SIRscript_output.eps')
